tests/integration/pilot/traffic_shifting_test.go

```diff
@@ -116,21 +119,18 @@ func TestTrafficShifting(t *testing.T) {
 		})
 }
 
-func echoConfig(ns namespace.Instance, name string) echo.Config {
-	return echo.Config{
-		Service:   name,
-		Namespace: ns,
-		Ports: []echo.Port{
-			{
-				Name:     "http",
-				Protocol: protocol.HTTP,
-				// We use a port > 1024 to not require root
-				InstancePort: 8090,
-			},
-		},
-		Subsets: []echo.SubsetConfig{{}},
-		Pilot:   p,
+// Wrapper to initialize instance with ServicePort without affecting other tests
+// If ServicePort is set to InstancePort, tests such as TestDescribe would fail
+func echoVMConfig(ns namespace.Instance, name string, vmImage ...string) echo.Config {
+	image := ""
+	if len(vmImage) > 0 {
+		image = vmImage[0]
 	}
+	config := echoConfig(ns, name)
+	config.DeployAsVM = image != ""
+	config.VMImage = image
+	config.Ports[0].ServicePort = config.Ports[0].InstancePort
```

Comment:
nit: comment that this is a hack to workaround a bug in the existing impl

Reply:
I will update! Thanks!